### PR TITLE
added more examples

### PR DESCRIPTION
**Pull request by wouterdb**
- moved the existing example into the lsm folder
- Added readme's
- Added two variations of an example 

# Questions for the reviewers:

- What do we intend to do testing-wise in this repo? (just a simple compile of the project or more?)
- Is this clear?

### CODE REVIEW

**Review by bartv — COMMENTED** [review]
I am not sure creating multiple levels will improve things. I do think it is a good idea to add lsm to it, like lsm-external-allocation

File: README.md [hunk 1]
```diff
@@ -1,2 +1,7 @@
 # examples
 Example modelling code
+
+
+# Folders
+
+1. LSM: examples related to Lifecycle and Service Management, these examples require a commercial license.
```
Comment (edvgui) [inline README.md:7]:
```suggestion
1. [LSM](lsm/README.md): examples related to Lifecycle and Service Management, these examples require a commercial license.
```

**Review by edvgui — COMMENTED** [review]
I think testing what we are trying to show (so more than a simple model compile) would be nice.  Not sure how we should structure those in the ci though.  One Jenkins file for the whole repository would probably grow really big.  Unless our tests are standard enough.

Looks clear to me.  As a tradeoff between folder and prefix, what about keeping the folders (make navigation on github easier if you know where to find it, but finding it the first time less easy), but add an index (with links) in the main readme that doesn't follow the folder structure (so with prefixes for example) to make it easier finding it the first time.

**Comment by bartv** [comment]
> I think testing what we are trying to show (so more than a simple model compile) would be nice. Not sure how we should structure those in the ci though. One Jenkins file for the whole repository would probably grow really big. Unless our tests are standard enough.
> 
> Looks clear to me. As a tradeoff between folder and prefix, what about keeping the folders (make navigation on github easier if you know where to find it, but finding it the first time less easy), but add an index (with links) in the main readme that doesn't follow the folder structure (so with prefixes for example) to make it easier finding it the first time.

For testing I would say the each project folder should contain a test.sh and that is it

File: README.md (hunk 1 above)
Comment (wouterdb) [reply README.md]: applied in main index.

**Comment by wouterdb** [comment]
@bartv a `test.sh` that assumes inmanta is already installed or not?

**Comment by bartv** [comment]
> @bartv a `test.sh` that assumes inmanta is already installed or not?

yes, we have to define something like:

- latest iso release is installed and compiler is available

**Comment by bartv** [comment]
> > @bartv a `test.sh` that assumes inmanta is already installed or not?
> 
> yes, we have to define something like:
> 
>     * latest iso release is installed and compiler is available

The goal for me is not that others can easily run it but that it does not bit root because for example a deprecated syntax or feature is used

File: README.md [hunk 2]
```diff
@@ -10,8 +10,8 @@ To learn LSM, the first thing to look at is the documentation: [here](https://do
 
 This repo has the following additional examples.
 
-1. `lsm-external-allocation`: example on how to allocate and de-allocate attributes from an external inventory
-2. `lsm-from-inventory-[a,b]`: example on how to load all attributes of a service from an external inventory
+1. [lsm-external-allocation](lsm-external-allocation/README.md): example on how to allocate and de-allocate attributes from an external inventory
+2. [lsm-from-inventory-](lsm-from-inventory-a/README.md): example on how to load all attributes of a service from an external inventory
```
Comment (edvgui) [inline README.md:14]:
```suggestion
2. [lsm-from-inventory-a](lsm-from-inventory-a/README.md): example on how to load all attributes of a service from an external inventory
3. [lsm-from-inventory-b](lsm-from-inventory-b/README.md): another example on how to load all attributes of a service from an external inventory
```

**Review by edvgui — CHANGES_REQUESTED** [review]

**Review by bartv — APPROVED** [review]

**Review by edvgui — APPROVED** [review]

**Pull request merged**